core/src/main/java/com/linecorp/armeria/client/ClientFactory.java

```diff
@@ -89,6 +90,11 @@ static void closeDefault() {
     ReleasableHolder<EventLoop> acquireEventLoop(Endpoint endpoint);
 
     /**
+     * Returns the {@link Metrics} that collects various stats.
+     */
+    Metrics metrics();
```

Comment:
If you have an access to `RequestContext`, then you should use `RequestContext.metrics()` which returns the same `Metrics` instance with `ClientFactory.metrics()`. If you are outside of the request execution, then you will probably have to use `ClientFactory.metrics()` because you do not have a reference to any `RequestContext`.